src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -1147,6 +1152,7 @@ BOOL HeapWalkHelper(Object * pBO, void * pvContext)
 
     if (pProfilerWalkHeapContext->fProfilerPinned)
     {
+#if defined(GC_PROFILING)
```

Comment:
This ifdef can start 2 lines earlier so that it includes `if (pProfilerWalkHeapContext->fProfilerPinned)`.